weather.py

```diff
@@ -0,0 +1,428 @@
+                     = self.average_per_month(date[0], month, avg_max_temp,
+                                              count_max_temp, avg_min_temp,
+                                              count_min_temp, avg_humidity, count_humidity)
+
+            if not exists:
+                raise Exception("Files for this year are not in the system.")
+
+        else:
+            mon = int(date[1])  # Getting month in word form
+            if 0 < mon < 13:  # Validate month is in between 1-12
+                month = self.months[mon - 1]
+
+                _, avg_max_temp, count_max_temp, avg_min_temp, \
+                    count_min_temp, avg_humidity, count_humidity \
+                    = self.average_per_month(date[0], month, avg_max_temp,
+                                             count_max_temp, avg_min_temp,
+                                             count_min_temp, avg_humidity, count_humidity)
+
+        results = dict()
+
+        # Storing average of results obtained in class level dictionaries
+        results["avgmaxtemp"] = avg_max_temp / count_max_temp
+        results["avgmintemp"] = avg_min_temp / count_min_temp
+        results["avghumid"] = avg_humidity / count_humidity
+
+        return results
+
+    def calculate_chart_form_data_helper(self, dic, results):
+
+        if dic["Max TemperatureC"] != "":
+            results["max_temp"].append(
+                int(dic["Max TemperatureC"]))  # adding to list in dictionary to print them all
+
+        if dic["Min TemperatureC"] != "":
+            results["min_temp"].append(
+                int(dic["Min TemperatureC"]))
+
+        return results
+
+    def calculate_chart_form_data(self, date):
+        results = dict()
+        results["max_temp"] = []
+        results["min_temp"] = []
+
+        mon = int(date[1])  # Getting month in word form
+        if 0 < mon < 13:  # Validate month is in between 1-12
+            file = self.files_path_prefix + date[0] + "_" + self.months[mon - 1] + ".txt"
+            reader = CSVReader(file)
+
+            dictionary = reader.get_line()
+            while dictionary:
+                results = self.calculate_chart_form_data_helper(dictionary, results)
+                dictionary = reader.get_line()
+
+            return results
+
+    # Method that calculates the read data
+    def argument_handler(self, curr, argument_type):
+        date = curr.split('/')
+        is_valid = self.check_input_validity(date, argument_type)
+
+        result = dict()
+
+        if is_valid:
+            if argument_type == "e":
+                result = self.calculate_highest(date)
+            elif argument_type == "a":
+                result = self.calculate_average(date)
+            else:
+                result = self.calculate_chart_form_data(date)
+
+        # Function that prints report according to the mode provided
+        self.print_report(argument_type, result)
+        result.clear()
+
```

Comment:
def handle_highest(current_argument):
    date = current_argument.split('/')
    is_valid = self.validate_date(date, argument_type)
    if not is_valid:
       return

   result = self.calculate_highest(date)
   self.generate_report_highest(result)